Rubberduck.Inspections/Concrete/ShadowedDeclarationInspection.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Rubberduck.Common;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Rubberduck.Parsing.Symbols;
+using Rubberduck.Parsing.VBA;
+using Rubberduck.UI;
+using Rubberduck.VBEditor.SafeComWrappers;
+
+namespace Rubberduck.Inspections.Concrete
+{
+    public sealed class ShadowedDeclarationInspection : InspectionBase
+    {
+        public ShadowedDeclarationInspection(RubberduckParserState state) : base(state)
+        {
+        }
+
+        public override CodeInspectionType InspectionType { get; } = CodeInspectionType.CodeQualityIssues;
+
+        public override IEnumerable<IInspectionResult> GetInspectionResults()
+        {
+            var builtInEventHandlers = State.DeclarationFinder.FindEventHandlers().ToHashSet();
+
+            var issues = new List<IInspectionResult>();
+
+            var allUserProjects = UserDeclarations.OfType(DeclarationType.Project).Cast<ProjectDeclaration>();
+
+            foreach (var userProject in allUserProjects)
+            {
+                var referencedProjectIds = userProject.ProjectReferences.Select(reference => reference.ReferencedProjectId).ToHashSet();
+
+                var userDeclarations = UserDeclarations.Where(d => d.ProjectId == userProject.ProjectId);
+
+                // User has no control over build-in event handlers or their parameters, so we skip them
+                userDeclarations = userDeclarations.Where(d => !DeclarationIsPartOfBuiltInEventHandler(d, builtInEventHandlers));
+
+                foreach (var declaration in userDeclarations)
+                {
+                    var shadowedDeclaration = State.AllDeclarations.FirstOrDefault(d =>
+                        referencedProjectIds.Contains(d.ProjectId) && d.IdentifierName == declaration.IdentifierName && DeclarationCanBeShadowed(d, declaration));
```

Comment:
Please split this clause at the `&&`s to shorten the line and make it easier to read